client/src/pages/Login/LoginPage.tsx

```diff
@@ -57,8 +64,12 @@ const LoginPage = ({ loggedIn, setLoggedIn }: LoginPageProps): JSX.Element => {
 
   checkloginCallback();
 
-  // Redirect if authentication is done:
-  if (loggedIn) return <Redirect to="/" />;
+  // Redirect if authentication is done
+  if (loggedIn) {
+    const prevPath = location.state?.prevPath ?? "/";
```

Comment:
If user is coming back from logging in through oAuth, `prevPath` is undefined.

Reply:
https://developers.google.com/identity/protocols/oauth2/javascript-implicit-flow#:~:text=it%20is%20requesting.-,state,-Recommended
We can fix that - google oauth provides a way to maintain a bit of state when doing auth.